feat(curriculum): add string method workshop (#66936)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/blocks/workshop-employee-profile-validator/69e76d1997661be7f0040e2d.md

---
id: 69e76d1997661be7f0040e2d
title: Step 15
challengeType: 20
dashedName: step-15
---

# --description--

> The `.split(separator)` method splits a string into a list of substrings at each occurrence of the separator. For example:

Reviewer: ```suggestion
The `.split(separator)` method splits a string into a list of substrings at each occurrence of `separator`. For example:
```

```py
fruits = 'apple, banana, cherry'
parts = fruits.split(', ')
print(parts)  # Output: ['apple', 'banana', 'cherry']
```

> Create a variable `full_address` and assign it `'123 Main Street, Springfield, IL'`. Then create `address_parts` by splitting on `', '`.

Reviewer: ```suggestion
Create a variable `full_address` and assign it `'123 Main Street, Springfield, IL'`. Then create `address_parts` by splitting on a comma followed by a single space.
```

> Then print `address_parts` with the prefix `'Address parts: '`. Since `address_parts` is a list, you will need to convert it to a string using `str()` before concatenating it with the prefix.

Reviewer: ```suggestion
Then print the result of concatenating the string `Address parts: ` and the `address_parts` variable. Since `address_parts` is a list, you will need to convert it to a string using `str()`.
```

# --hints--

You should create a variable named `full_address`.

```js
({ test: () => assert(runPython(`_Node(_code).has_variable('full_address')`)) })
```

You should assign `'123 Main Street, Springfield, IL'` to `full_address`.

```js
({ test: () => assert(runPython(`
node = _Node(_code).find_variable('full_address')
not node.is_empty() and node.is_equivalent("full_address = '123 Main Street, Springfield, IL'")
`)) })
```

You should create a variable named `address_parts`.

```js
({ test: () => assert(runPython(`_Node(_code).has_variable('address_parts')`)) })
```

> You should use `.split(', ')` on `full_address` to assign `address_parts`.

Reviewer 2: Missing "a value", unlike the equivalent hints in the surrounding steps.

```suggestion
You should use `.split(', ')` on `full_address` to assign `address_parts` a value.
```

> You should use `.split(', ')` on `full_address` to assign `address_parts` a value.

Reviewer: ```suggestion
You should use `.split(', ')` on `full_address` and assign the result to the `address_parts` variable.
```

```js
({ test: () => assert(runPython(`
node = _Node(_code).find_variable('address_parts')

> not node.is_empty() and node.is_equivalent("address_parts = full_address.split(', ')")

Reviewer: ```suggestion
_Node(_code).find_variable('address_parts').is_equivalent("address_parts = full_address.split(', ')")
```

`)) })
```

> You should print `address_parts` with the prefix `'Address parts: '`. Don't forget to use `str()` to convert `address_parts` when printing.

Reviewer: ```suggestion
You should print the result of concatenating the string `Address parts: ` and the `address_parts` variable.. Don't forget to use `str()` to convert `address_parts` when printing.
```

---
id: 69e76d1997661be7f0040e2d
title: Step 15
challengeType: 20
dashedName: step-15
---

# --description--

The `.split(separator)` method splits a string into a list of substrings at each occurrence of `separator`. For example:

```py
fruits = 'apple, banana, cherry'
parts = fruits.split(', ')
print(parts)  # Output: ['apple', 'banana', 'cherry']
```

> Create a variable `full_address` and assign it `123 Main Street, Springfield, IL`. Then create `address_parts` by splitting on a comma followed by a single space.

Reviewer 2: Same two points as step 14, and naming `full_address` as the thing being split lets the second sentence stand on its own.

```suggestion
Create a variable named `full_address` and assign it the string `123 Main Street, Springfield, IL`. Then create a variable named `address_parts` by splitting `full_address` on a comma followed by a single space.
```
